src/ejabberd_sm.erl

```diff
@@ -499,18 +496,13 @@ init([]) ->
                           undefined -> {mnesia, []};
                           Value -> Value
                       end,
-    store_backend(Backend),
-
+    mongoose_backend:init(global, ?MODULE, [], [{backend, Backend}]),
```

Comment:
Just to replicate what all other modules do, here we could be doing
```suggestion
    ejabberd_sm_backend:start(Opts),
```
And `ejabberd_sm_backend` can do all the `mongoose_backend:init` together with the `ejabberd_sm_mnesia:start` callback and so on.